picard/acoustid/__init__.py

```diff
@@ -77,15 +77,15 @@ def _on_lookup_finished(self, next_func, file, document, http, error):
                             parsed_recording = parse_recording(recording)
                             if parsed_recording is not None:
                                 recording_list.append(parsed_recording)
-                        log.debug("AcoustID: Lookup successful for '%s'", file.filename)
+                        log.debug("AcoustID: Lookup successful for '%s'", file.filename, domains='acoustid')
             else:
                 mparms = {
                     'error': document['error']['message'],
                     'filename': file.filename
                 }
                 log.error(
                     "AcoustID: Lookup error for '%(filename)s': %(error)r" %
-                    mparms)
+                    mparms, domains='acoustid')
```

Comment:
Instead of manually specifying the domains, we can use the built in logging module from python stdlib and take advantage of its attributes https://docs.python.org/3/library/logging.html#logrecord-attributes

This will save us all the overhead and the repetitions

Reply:
I'd say we can simply drop the domains part and just keep the UI additions and maybe we can continue this in another PR

Reply:
> Instead of manually specifying the domains, we can use the built in logging module from python stdlib and take advantage of its attributes https://docs.python.org/3/library/logging.html#logrecord-attributes
> 
> This will save us all the overhead and the repetitions

I don't see how LogRecord provides this feature, as is.
Using module is not what i want, since, in the same module, i want to be able to specify the domain of the log line. Filtering on module can be good, but that's not my goal.
In fact, to implement this you need to subclass `logging.Logger`, something like that:

```python

def _onestr2set(domains):
    if not domains:
        return set()
    if isinstance(domains, str):
        # this is a shortcut: allow one to pass a domains as sole
        # argument, but use it as first argument of set
        return set((domains,))
    else:
        return set(domains)


class CustomLogger(logging.Logger):

    def _log(self, level, msg, args, exc_info=None, extra=None):
        if extra is None:
            extra = { 'domains': set() }
        super(CustomLogger, self)._log(level, msg, args, exc_info, extra)


logging.setLoggerClass(CustomLogger)

def debug(message, *args, domains=None):
    domains = _onestr2set(domains)
     root.debug(message, *args, extra={'domains': domains})


def info(message, *args, domains=None):
    domains = _onestr2set(domains)
    root.info(message, *args, extra={'domains': domains})


def warning(message, *args, domains=None):
    domains = _onestr2set(domains)
    root.warning(message, *args, extra={'domains': domains})


def error(message, *args, domains=None):
    domains = _onestr2set(domains)
    root.error(message, *args, extra={'domains': domains})

root = logging.getLogger('picard')
root.setLevel(logging.DEBUG)

handler = logging.StreamHandler()
bf = logging.Formatter('{asctime} {levelname:8s} {message} {domains}', style='{')
handler.setFormatter(bf)
root.addHandler(handler)

```